added isPrime plugin to plugins.json

lvonkacsoh opened the pull request:
I hope everything's okay - first time creating a PR 😅

jjw24 commented:
hey thanks for the add, you forgot to put the repo as public.

lvonkacsoh commented:
You are right! I fixed it

jjw24 commented on plugins.json:
```diff
@@ -201,6 +201,21 @@
+        "Language": "python",
+        "Website": "https://github.com/lvonkacsoh/Flow.Launcher.Plugin.IsPrime",
+        "IcoPath": "assets\\icon.png",
+        "ExecuteFileName": "main.py",
```
could we please remove the attributes 'ActionKeyword' and 'ExecuteFileName', they are not needed.

jjw24 commented on plugins.json:
```diff
@@ -201,6 +201,21 @@
         "Website": "https://github.com/JohnTheGr8/Flow.Plugin.Github",
         "UrlDownload": "https://github.com/JohnTheGr8/Flow.Plugin.Github/releases/download/v1.2.2/Flow.Plugin.Github.zip",
         "UrlSourceCode": "https://github.com/JohnTheGr8/Flow.Plugin.Github/tree/main",
+        "E-tag": "W/\"6f4c4aa08e4286539f224e2af0cc37032c7a4c141e12ce7b65bd39639dab2695\""
```
could you remove this E-tag and move this bottom e-tag: "E-tag": "W/\"b4aeb7a3c265e59c4a0736774bbaff98c0827b5758a4f9790d280d32d8b6932b\"" up here please.

E-Tag for the plugin be generated by the repo's CI automatically, so you dont need to add one for yours.

jjw24 commented:
I think it would be good if you upload the built package in Releases section of the repo, this will allow the manifest's CI to auto update its manifest with newer versions when you upload future releases.

![image](https://user-images.githubusercontent.com/26427004/115109860-ae7c1400-9fbb-11eb-88cb-0cef56b6b8ce.png)

lvonkacsoh commented:
> I think it would be good if you upload the built package in Releases section of the repo, this will allow the manifest's CI to auto update its manifest with newer versions when you upload future releases.

I'm sorry, I am quiet inexperienced with GitHub so I didn't fully get what I should do. May I ask you to explain further how I can improve the plugin so your CI can update its manifest automatically?

taooceros commented:
> I'm sorry, I am quiet inexperienced with GitHub so I didn't fully get what I should do. May I ask you to explain further how I can improve the plugin so your CI can update its manifest automatically?

Don't worry too much. You don't have to build the whole package since the plugin is written in python. All you need to do is to create a zip file that include everything of the plugin and attach that zip to the Release in github.

jjw24 commented:
Yeah no problem at all. The Releases page is one way of releasing your code for people to download, eg. if you look at Flow's release page, you can download the executable right, so for your plugin you can manually download the artifact from the url https://github.com/lvonkacsoh/Flow.Launcher.Plugin.IsPrime/archive/refs/tags/v1.0.zip and upload it to your initial release version. You can click on edit on the the release page to upload the artifact.

Like this:
![image](https://user-images.githubusercontent.com/26427004/115110467-f3ee1080-9fbe-11eb-9bf9-419e49807b66.png)

We have a CI in this repo to automatically scan the plugin repos and then up the version and url so users will be able to download your newly published version and you don't have to make a PR here every time you release a new version. The code for the CI to update automatically is here:
https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest/blob/main/updateChecker.py

In the future if you are going to do more releases, you can consider doing a CI to have Github automatically package up and release your plugin.

lvonkacsoh commented:
Thank you guys for the clarification and your patience! I added the artefact .zip file to the release and updated the manifest. Hopefully everything is good now :)

jjw24 commented:
All good to go, merged. Thanks for sharing
